server/src/main/java/org/apache/gravitino/server/web/rest/TagOperations.java

```diff
@@ -0,0 +1,467 @@
+  @Timed(name = "alter-tag." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "alter-tag", absolute = true)
+  public Response alterTag(
+      @PathParam("metalake") String metalake,
+      @PathParam("tag") String name,
+      TagUpdatesRequest request) {
+    LOG.info("Received alter tag request for tag: {} under metalake: {}", name, metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            request.validate();
+
+            TagChange[] changes =
+                request.getUpdates().stream()
+                    .map(TagUpdateRequest::tagChange)
+                    .toArray(TagChange[]::new);
+            Tag tag = tagManager.alterTag(metalake, name, changes);
+
+            LOG.info("Altered tag: {} under metalake: {}", name, metalake);
+            return Utils.ok(new TagResponse(DTOConverters.toDTO(tag, Optional.empty())));
+          });
+    } catch (Exception e) {
+      return ExceptionHandlers.handleTagException(OperationType.ALTER, name, metalake, e);
+    }
+  }
+
+  @DELETE
+  @Path("{tag}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "delete-tag." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "delete-tag", absolute = true)
+  public Response deleteTag(@PathParam("metalake") String metalake, @PathParam("tag") String name) {
+    LOG.info("Received delete tag request for tag: {} under metalake: {}", name, metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            boolean deleted = tagManager.deleteTag(metalake, name);
+            if (!deleted) {
+              LOG.warn("Failed to delete tag {} under metalake {}", name, metalake);
+            } else {
+              LOG.info("Deleted tag: {} under metalake: {}", name, metalake);
+            }
+
+            return Utils.ok(new DropResponse(deleted));
+          });
+    } catch (Exception e) {
+      return ExceptionHandlers.handleTagException(OperationType.DELETE, name, metalake, e);
+    }
+  }
+
+  @GET
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "list-object-tags." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "list-object-tags", absolute = true)
+  public Response listTagsForMetdataObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      @QueryParam("details") @DefaultValue("false") boolean verbose) {
+    LOG.info(
+        "Received list tag {} request for object type: {}, full name: {} under metalake: {}",
+        verbose ? "infos" : "names",
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            MetadataObject object = MetadataObjects.parse(fullName, toType(type));
+
+            if (verbose) {
```

Comment:
I suggest we combine the logic for getting tag information and tag name as they both need to the full tag information from underlying storage